lib/student.rb

```diff
@@ -0,0 +1,17 @@
+class Student
+    attr_reader :name, :age, :scores
+    def initialize(student_data)
+        @name = student_data[:name]
+        @age = student_data[:age]
+        @scores = student_data.fetch(:scores, [])
+    end
+
+    def log_score(score)
+        @scores << score
+    end
+
+    def grade
+        return 0 if @scores.empty?
```

Comment:
This is really good here, checking to see what happens if there are no scores. Great solution for this too.